### PR TITLE
Adds helpers for finding martial arts types

### PR DESCRIPTION
**Pull request by MadmanMartian**
:cl: 
code: Adds helpers for finding if a user has a martial art type
/:cl:

[why]: requested in coderbus, as somebody is making a new martial art or something.

### CODE REVIEW

**Comment by tortellinitony** [comment]
What if we make martial arts a bitwise variable and then you can have all of them at once

**Comment by MadmanMartian** [comment]
Could do that, but that'd be out of scope

File: code/__DEFINES/melee.dm [hunk 1]
```diff
@@ -0,0 +1,10 @@
+//Martial arts defines
+
+#define MARTIALART_BOXING "boxing"
+#define MARTIALART_WRESTLING "wrestling"
+#define MARTIALART_SLEEPINGCARP "sleeping carp"
+#define MARTIALART_PSYCHOBRAWL "psychotic brawling"
```
Comment (Cyberboss) [inline code/__DEFINES/melee.dm]: These are display names and shouldn't be lowercased

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/__DEFINES/melee.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/__DEFINES/melee.dm]: Changes made, they're now used in an ID variable and never seen by the mortal eye.

File: code/game/objects/items/granters.dm [hunk 2]
```diff
@@ -341,11 +341,9 @@
 	if(!martial)
 		return
 	var/datum/martial_art/MA = new martial
-	if(user.mind.martial_art)
-		for(var/datum/martial_art/knownmartial in user.mind.martial_art)
-			if(knownmartial.type == MA.type)
-				to_chat(user,"<span class='warning'>You already know [martialname]!</span>")
-				return
+	if(user.mind.has_martialart(MA.name))
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
 	if(user.mind.has_martialart(MA.id))
```

File: code/game/objects/items/granters.dm [hunk 3]
```diff
@@ -341,11 +341,9 @@
...
+		to_chat(user,"<span class='warning'>You already know [martialname]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
 		to_chat(user,"<span class='warning'>You already know [MA.name]!</span>")
```

Comment (Cyberboss) [reply code/game/objects/items/granters.dm]: the `martialname` var can be removed entirely too i think

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/granters.dm (hunk 3 above)
Comment (MadmanMartian) [reply code/game/objects/items/granters.dm]: Looking at it, some of the martialname stuff is for fluff, but necessary fluff. Like the CQC book has martial name as close quarters combat

**Pull request merged**